ENGAGEHF/Dashboard/Notifications/Views/NotificationRow.swift

```diff
@@ -0,0 +1,111 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SpeziViews
+import SwiftUI
+
+
+struct NotificationRow: View {
+    @Binding var notification: Notification
+    @State private var isExpanded = false
+    
+    
+    var body: some View {
+        VStack(alignment: .leading) {
+            HStack(alignment: .center) {
+                Text(notification.type)
+                    .font(.subheadline)
+                    .foregroundStyle(.secondary)
+                Spacer()
+                XButton(notification: $notification)
+            }
+            Divider()
+            Text(notification.title)
+                .multilineTextAlignment(.leading)
+                .padding(.bottom, 10)
+            if isExpanded {
+                Text(notification.description)
+                    .multilineTextAlignment(.leading)
+                    .font(.footnote)
+            } else {
+                LearnMoreButton(isExpanded: $isExpanded)
+            }
+        }
+            .onDisappear {
+                isExpanded = false
+            }
+    }
+    
+    
+    private struct XButton: View {
+        @Environment(NotificationManager.self) private var notificationManager
+        @Binding var notification: Notification
+        
+        
+        var body: some View {
+            AsyncButton(
+                action: {
+                    let indx = notificationManager.notifications.firstIndex { $0.id == notification.id }
+                    
+                    if let indx: Int {
+                        await notificationManager.markComplete(at: IndexSet(integer: indx))
+                    }
```

Comment:
As we are matching ids to index paths and then back to ids in the notification manager I would suggest to transform this logic here and in the notification manager to actually work based on the notification id and not using the indirection using the index identification.